src/local/room_coordinate.rs

```diff
@@ -0,0 +1,113 @@
+use std::convert::{TryFrom, TryInto};
+use std::fmt;
+
+pub const ROOM_SIZE: u8 = 50;
+
+#[derive(Debug, Clone, Copy)]
+pub struct OutOfBoundsError(u8);
+
+impl fmt::Display for OutOfBoundsError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Out of bounds coordinate: {}", self.0)
+    }
+}
+
+#[inline]
+pub fn xy_to_linear_index(xy: RoomXY) -> usize {
+    ((xy.x.0 as usize) * (ROOM_SIZE as usize)) + (xy.y.0 as usize)
+}
+
+#[inline]
+pub fn linear_index_to_xy(idx: usize) -> RoomXY {
+    RoomXY {
```

Comment:
Maybe I should check that idx is a valid linear index, then use the unchecked versions below? Avoids missing an overflow after the downcast.